docs/whatsnew/7.0.rst

```diff
@@ -436,6 +437,32 @@ simple example might be:
     >>> stats.biweight_location(), stats.biweight_scale()  # doctest: +FLOAT_CMP
     (np.float64(67.98055399699436), np.float64(64.82889460022386))
 
+.. _whatsnew-7.0-masked-coordinates:
+
+Support for masks in ``coordinates``
+====================================
+
+Initial support for masks has been implemented for |SkyCoord|, as well as for
+coordinate frames and coordinate representations.
+
+Like more generally for |Masked| instances, one can access the mask using
+`~astropy.coordinates.SkyCoord.mask` and check whether the underlying data are
+masked using `~astropy.coordinates.SkyCoord.masked`. Like for all masked
+classes, one can access underlying unmasked data using
+`~astropy.utils.masked.MaskableShapedLikeNDArray.unmasked` and one can get a
+new instance with masked data replaced with a fill value using
+:meth:`~astropy.utils.masked.MaskableShapedLikeNDArray.filled`.
```

Comment:
When I try and use this, the `filled` object is still masked. Is this intended? It's not what I expect. Using the example from the main docs....

```python
In [23]: gcrs.filled(gcrs2[1])
Out[23]: 
<SkyCoord (GCRS: obstime=J2000.000, obsgeoloc=(0., 0., 0.) m, obsgeovel=(0., 0., 0.) m / s): (ra, dec, distance) in (deg, deg, kpc)
    [(15.00054403, 2.99988395, 0.1), (        ———,        ———, ———)]>
```

```python
In [25]: gcrs.filled(gcrs2[1]).unmasked
Out[25]: 
<SkyCoord (GCRS: obstime=J2000.000, obsgeoloc=(0., 0., 0.) m, obsgeovel=(0., 0., 0.) m / s): (ra, dec, distance) in (deg, deg, kpc)
    [(15.00054403, 2.99988395, 0.1), (30.00201927, 3.99996188, 1. )]>
```

Reply:
Thanks for noticing that! The issue arises because the fill value you are using is masked, and that the causes the output to become masked again as well (basically, `.filled()` just does `result = self.unmasked.copy(); result[self.mask] = fill_value`-- if you do `gcrs.filled(gcrs[1].unmasked)` it works properly.

Now this may explain it, but of course does not make it any less expected. What do you think would be best? Raising if `fill_value` is masked? Or a warning? Or ignoring the mask?

FWIW, for an actual array (or Quantity), the mask on the fill value is simply ignored, and I see that this holds for `Time` too -- so maybe I should ensure that happens here too... I'll look a bit later.

Reply:
I’m relaxed about this remaining as it is. Once you’ve seen it, it’s easy for the user to workaround in the manner you suggest. 

If you have time we could ignore the mask on the fill value (I like consistency) but it should prevent merge in time for the big freeze

Reply:
My last commit ensured the mask on the fill value is ignored; I agree it is best to be consistent, at least for now.